cmake/benchmark.cmake

```diff
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+set(BENCHMARK_ENABLE_GTEST_TESTS OFF CACHE BOOL "Turn off gTest in gBenchmark")
```

Comment:
just use `set(BENCHMARK_ENABLE_GTEST_TESTS OFF)`
Also, setting this is only relevant if the provider is "module" so you should move this under the if  branch for "module". (For "package", the change will have no effect as that branch is for a pre-installed version of benchmark and no build is involved).

Reply:
the reason for cache is that somehow CMake does not propagate non-cached variables to subdirectories. Without the CACHE keyword, it still tried to build gtest

Reply:
we should still move under `if("${gRPC_BENCHMARK_PROVIDER}" STREQUAL "module")`, see my comment above.

Reply:
addressed